bbs/ssh.cpp

```diff
@@ -43,190 +43,388 @@
 #include "core/log.h"
 #include "core/net.h"
 #include "core/os.h"
+#include "core/strings.h"
+#include "bbs/bbs.h"
+
+#include <libssh/libssh.h>
+#include <libssh/server.h>
 
 using std::string;
 using std::thread;
 using std::unique_ptr;
 
+using namespace wwiv::core;
+using namespace wwiv::strings;
+
 namespace wwiv {
 namespace bbs {
 
 struct socket_error: public std::runtime_error {
   socket_error(const std::string& message): std::runtime_error(message) {}
 };
 
-static constexpr char WWIV_SSH_KEY_NAME[] = "wwiv_ssh_server";
-#define RETURN_IF_ERROR(s) { if (!cryptStatusOK(s)) return false; }
-#define OK(s) cryptStatusOK(s)
+Keys::Keys() {
 
-static bool ssh_once_init() {
-  // This only should be called once.
-  int status = cryptInit();
-  return OK(status);
 }
 
-bool Key::Open() {
-  static bool once = ssh_once_init();
-  CRYPT_KEYSET keyset;
-  int status = cryptKeysetOpen(&keyset, CRYPT_UNUSED, CRYPT_KEYSET_FILE, filename_.c_str(), CRYPT_KEYOPT_NONE);
-  RETURN_IF_ERROR(status);
+// Check that at least one private key exists
+bool Keys::Exist() {
```

Comment:
I wonder if we could support both key classes (i.e. CryptlibKeys, and LibSshKeys with a common base class, ditto for the SSH implementation class).  It may be a little while before someone wants to switch if there's compatibility issues?  WDYT? Did you consider that? (I'm just thinking out loud)

Reply:
AFAIK, the current ssh works only with syncterm, which is a non-starter for many people.  That's a bit ironic because syncterm is still one of the best terminals to use from a visual/functional perspective.  Regardless, I think it's a better direction to make WWIV work more generally with more common OpenSSH, but I'd hate to see syncterm stop working as a result (Macs don't have many good options besides syncterm).